Update the flutter version and other packages

The app no longer works. Most probably it's an issue with one (or more) package versions.
Pubspec needs to be updated.

Hi, I tried to run the app on my device but i am getting some errors.

`[+11781 ms] [+13695 ms] ../../flutter/.pub-cache/git/flutter_duration_picker-285ff19ad8b440fb6b8727ea849b7ce679d67c75/lib/flutter_duration_picker.dart:104:18: Error:
The getter 'display3' isn't defined for the class 'TextTheme'.
[   +2 ms] [   +2 ms]  - 'TextTheme' is from 'package:flutter/src/material/text_theme.dart' ('../../flutter/packages/flutter/lib/src/material/text_theme.dart').
[   +1 ms] [        ] Try correcting the name to the name of an existing getter, or defining a getter or field named 'display3'.
[        ] [        ]                 .display3
[        ] [        ]                  ^^^^^^^^
[        ] [   +3 ms] ../../flutter/.pub-cache/git/flutter_duration_picker-285ff19ad8b440fb6b8727ea849b7ce679d67c75/lib/flutter_duration_picker.dart:117:48: Error: The
getter 'body1' isn't defined for the class 'TextTheme'.
[   +2 ms] [        ]  - 'TextTheme' is from 'package:flutter/src/material/text_theme.dart' ('../../flutter/packages/flutter/lib/src/material/text_theme.dart').       
[   +1 ms] [        ] Try correcting the name to the name of an existing getter, or defining a getter or field named 'body1'.
[   +1 ms] [        ]             style: Theme.of(context).textTheme.body1),
[   +1 ms] [        ]                                                ^^^^^
[   +1 ms] [   +3 ms] ../../flutter/.pub-cache/git/flutter_duration_picker-285ff19ad8b440fb6b8727ea849b7ce679d67c75/lib/flutter_duration_picker.dart:440:39: Error: The
getter 'subhead' isn't defined for the class 'TextTheme'.
[   +2 ms] [        ]  - 'TextTheme' is from 'package:flutter/src/material/text_theme.dart' ('../../flutter/packages/flutter/lib/src/material/text_theme.dart').       
[   +1 ms] [        ] Try correcting the name to the name of an existing getter, or defining a getter or field named 'subhead'.
[   +1 ms] [        ]     final TextStyle style = textTheme.subhead;
[        ] [        ]                                       ^^^^^^^
[+13373 ms] [+13404 ms] Persisting file store
[   +1 ms] [   +5 ms] Done persisting file store
[  +92 ms] [   +6 ms] Target kernel_snapshot failed: Exception
[   +1 ms]            #0      KernelSnapshot.build (package:flutter_tools/src/build_system/targets/common.dart:241:7)
[   +1 ms]            <asynchronous suspension>
[        ]            #1      _BuildInstance._invokeInternal (package:flutter_tools/src/build_system/build_system.dart:839:9)
[        ]            <asynchronous suspension>
[        ]            #2      Future.wait.<anonymous closure> (dart:async/future.dart)
[        ]            <asynchronous suspension>
[        ]            #3      _BuildInstance.invokeTarget (package:flutter_tools/src/build_system/build_system.dart:777:32)
[        ]            <asynchronous suspension>
[        ]            #4      FlutterBuildSystem.build (package:flutter_tools/src/build_system/build_system.dart:606:16)
[   +1 ms]            <asynchronous suspension>
[   +1 ms]            #5      AssembleCommand.runCommand (package:flutter_tools/src/commands/assemble.dart:317:32)
[   +1 ms]            <asynchronous suspension>
[        ]            #6      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1140:27)
[        ]            <asynchronous suspension>
[        ]            #7      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
[   +1 ms]            <asynchronous suspension>
[   +1 ms]            #8      CommandRunner.runCommand (package:args/command_runner.dart:209:13)
[        ]            <asynchronous suspension>
[        ]            #9      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:288:9)
[        ]            <asynchronous suspension>
[        ]            #10     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
[        ]            <asynchronous suspension>
[        ]            #11     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:236:5)
[        ]            <asynchronous suspension>
[        ]            #12     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
[        ]            <asynchronous suspension>
[        ]            #13     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
[        ]            <asynchronous suspension>
[        ]            #14     main (package:flutter_tools/executable.dart:94:3)
[        ]            <asynchronous suspension>
[        ] [   +5 ms] 
[        ]            #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
[        ]            #1      AssembleCommand.runCommand (package:flutter_tools/src/commands/assemble.dart:334:7)
[        ]            <asynchronous suspension>
[   +1 ms]            #2      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1140:27)
[        ]            <asynchronous suspension>
[        ]            #3      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
[        ]            <asynchronous suspension>
[        ]            #4      CommandRunner.runCommand (package:args/command_runner.dart:209:13)
[        ]            <asynchronous suspension>
[        ]            #5      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:288:9)
[        ]            <asynchronous suspension>
[        ]            #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
[        ]            <asynchronous suspension>
[        ]            #7      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:236:5)
[        ]            <asynchronous suspension>
[        ]            #8      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
[        ]            <asynchronous suspension>
[        ]            #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
[   +1 ms]            <asynchronous suspension>
[   +1 ms]            #10     main (package:flutter_tools/executable.dart:94:3)
[   +1 ms]            <asynchronous suspension>
[   +1 ms] [  +20 ms] "flutter assemble" took 28,092ms.
[ +157 ms] [ +257 ms] ensureAnalyticsSent: 256ms
[   +1 ms] [   +2 ms] Running shutdown hooks
[        ] [        ] Shutdown hooks complete
[        ] [        ] exiting with code 1
[ +203 ms] > Task :app:compileFlutterBuildDebug FAILED
[   +1 ms] FAILURE: Build failed with an exception.
[   +1 ms] * Where:
[   +1 ms] Script 'C:\Users\DELL\Documents\flutter\packages\flutter_tools\gradle\flutter.gradle' line: 1070
[   +1 ms] * What went wrong:
[   +1 ms] Execution failed for task ':app:compileFlutterBuildDebug'.
[   +1 ms] > Process 'command 'C:\Users\DELL\Documents\flutter\bin\flutter.bat'' finished with non-zero exit value 1
[   +1 ms] * Try:
[        ] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.        
[   +1 ms] * Get more help at https://help.gradle.org
[   +1 ms] BUILD FAILED in 39s
[   +1 ms] Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
[        ] Use '--warning-mode all' to show the individual deprecation warnings.
[   +1 ms] See https://docs.gradle.org/6.7.1/userguide/command_line_interface.html#sec:command_line_warnings
[        ] 1 actionable task: 1 executed
[ +675 ms] Running Gradle task 'assembleDebug'... (completed in 41.2s)
[   +8 ms] Exception: Gradle task assembleDebug failed with exit code 1
[   +2 ms] "flutter run" took 44,218ms.
[   +5 ms]
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:674:9)
           <asynchronous suspension>
           #2      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1140:27)
           <asynchronous suspension>
           #3      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #4      CommandRunner.runCommand (package:args/command_runner.dart:209:13)
           <asynchronous suspension>
           #5      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:288:9)
           <asynchronous suspension>
           #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:236:5)
           <asynchronous suspension>
           #8      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #10     main (package:flutter_tools/executable.dart:94:3)
           <asynchronous suspension>


[ +262 ms] ensureAnalyticsSent: 255ms
[   +2 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1`

--- a/lib/components/create_join_dialog.dart
+++ b/lib/components/create_join_dialog.dart
@@ -5,7 +5,7 @@ import 'package:beacon/components/hike_button.dart';
 import 'package:beacon/utilities/constants.dart';
 import 'package:beacon/view_model/home_view_model.dart';
 import 'package:flutter/material.dart';
-import 'package:flutter_duration_picker/flutter_duration_picker.dart';
+import 'package:duration_picker/duration_picker.dart';
 import 'package:google_maps_flutter/google_maps_flutter.dart';
 
 class CreateJoinBeaconDialog {

--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -1,10 +1,12 @@
 name: beacon
 description: A new Flutter project.
 
+publish_to: "none"
+
 version: 1.0.0+1
 
 environment:
-  sdk: '>=2.7.0 <3.0.0'
+  sdk: ">=2.7.0 <3.0.0"
 
 dependencies:
   flutter:
@@ -26,7 +28,7 @@ dependencies:
   provider: ^5.0.0
   uni_links: ^0.5.1
   geocoder: ^0.2.1
-  share: ^2.0.4 
+  share: ^2.0.4
   location: ^4.3.0
   rxdart: ^0.26.0
   flutter_polyline_points: ^1.0.0
@@ -35,8 +37,7 @@ dependencies:
   sliding_up_panel: ^2.0.0+1
   skeleton_text: ^1.0.0
   flutter_config: ^2.0.0
-  flutter_duration_picker:
-    git: https://github.com/bartektartanus/flutter_duration_picker
+  duration_picker: ^1.1.0+1
 
 dev_dependencies:
   flutter_test:

